clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2228,18 +2236,22 @@ void handleFailure(Throwable throwable) {
     @Override
     public DescribeConsumerGroupsResult describeConsumerGroups(final Collection<String> groupIds,
                                                                final DescribeConsumerGroupsOptions options) {
-        final KafkaFutureImpl<Map<String, KafkaFuture<ConsumerGroupDescription>>> resultFutures = new KafkaFutureImpl<>();
-        final Map<String, KafkaFutureImpl<ConsumerGroupDescription>> consumerGroupFutures = new HashMap<>(groupIds.size());
-        final ArrayList<String> groupIdList = new ArrayList<>();
-        for (String groupId : groupIds) {
-            if (!consumerGroupFutures.containsKey(groupId)) {
-                consumerGroupFutures.put(groupId, new KafkaFutureImpl<ConsumerGroupDescription>());
-                groupIdList.add(groupId);
+
+        final Map<String, KafkaFutureImpl<ConsumerGroupDescription>> futures = new HashMap<>(groupIds.size());
+        for (String groupId: groupIds) {
+            if (groupIdIsUnrepresentable(groupId)) {
```

Comment:
Good point.

Reply:
Hmm, interesting... so people can actually use an empty string as a group ID?

If I understand correctly, it seems like this point has been fixed, since `KafkaAdminClient#groupIdIsUnrepresentable` now allows empty strings (it only disallows null.)

Reply:
By the way, the decision to filter "unrepresentable" topic names rather than "invalid" ones was deliberate.  As much as possible, we want to have decisions about what names are or are not valid be a server-side decision.  That way it's easy to change the policy without rolling out new clients.

AdminClient has to filter out empty topic names only because the empty string has a special meaning in the RPC.  If we could, we'd let the server do it.  (Obviously null topic names also can't be represented on the wire either, and so must be filtered.)